src/modules/QtQuick/ColorAnimation.js

```diff
@@ -0,0 +1,132 @@
+const EMPTY_ANIMATION_COLOR = new QmlWeb.QColor();
+
+QmlWeb.registerQmlType({
+  module: "QtQuick",
+  name: "ColorAnimation",
+  versions: /.*/,
+  baseClass: "PropertyAnimation",
+  properties: {
+    from: "color",
+    to: "color",
+  }
+}, class {
+  constructor(meta) {
+    QmlWeb.callSuper(this, meta);
+
+    this.$at = 0;
+    this.$loop = 0;
+
+    QmlWeb.engine.$addTicker((...args) => this.$ticker(...args));
+    this.runningChanged.connect(this, this.$onRunningChanged);
+  }
+  $redoActions() {
+    this.$actions = [];
+    for (let i = 0; i < this.$targets.length; i++) {
+      for (const j in this.$props) {
+        const target = this.$targets[i];
+        const property = this.$props[j];
+        const from_color = this.from;
+        const to_color = this.to;
+        // Animations always take time in SequentialAnimation
+        // , regardless of the value from and to
+        const action = {
+          // from, to
+          target,
+          property,
+          from_color,
+          to_color
+        };
+        this.$actions.push(action);
+      }
+    }
+    this.$startLoop();
+  }
+  $startLoop() {
+    for (const i in this.$actions) {
+      const action = this.$actions[i];
+      const from_color = action.from_color ||
+        action.target[action.property] ||
+        EMPTY_ANIMATION_COLOR;
```

Comment:
Does it make sense to cache this in a separate variable? This isn't the place where the slowdown would be here.
1. This usecase (when neither `from` nor target property value) is specified is very unlikely.
2. This only happens on the beginning of each loop and saves at most of one `new` call.

I would just write `new QmlWeb.QColor()` here, I think that it wouldn't do any measurable difference, and that would keep the code a bit cleaner.